src/book/03-guides/04-general/04-writing-tests.mdx

```diff
@@ -0,0 +1,60 @@
+---
+section: Guides
+chapter: General
+title: How to Write Tests
+description: Information About Making Tests
+slug: /guides/general/writing-tests
+---
+
+This guide presents some general information about unit testing and tests in general.
+
+## What is a Unit Test?
+
+A unit test is a test of a small piece of a codebase - a unit. Unit tests are meant to test only that piece of code and they serve to validate its correctness. This contrasts with integration tests, which tests the interactions of pieces of code and their behaviour together. For the rest of this guide, we'll work with the following toy utility function
+
+```cpp
+[[nodiscard]] inline int add(const int a, const int b) noexcept {
```

Comment:
😆 I can't help but wonder if this would confuse some people.

Reply:
Kip and his love of nodiscard inline noexcept 😅

Reply:
honestly I am torn about this. It adds "needless" complexity, but at the same time I was thinking it should be code that I would actually consider approving. I guess that's maybe a stretch in itself though, because I would not approve a function to add two integers :joy: